sdk/src/util/driftClock/rpcConnectedClockSubscriber.ts

```diff
@@ -0,0 +1,63 @@
+import { Connection } from "@solana/web3.js";
+import { BaseClockSubscriber } from "./baseClockSubscriber";
+
+/**
+ * A clock subscriber that listens to the RPC connection and updates the clock accordingly.
+ */
+export class RpcConnectedClockSubscriber extends BaseClockSubscriber {
```

Comment:
i wonder if it would be easier on the RPCs to allow user to optionally pass in a `SlotSusbcriber`, then fall back to the current polling if not.

We also have `BlockhashSubscriber` too, which polls getLatestBlockhash and keeps a queue of the last x blockhash+blockheights, might be useful for this